compass-js-frontend/src/main/webapp/ui/view/editor/hierarchy.js

```diff
@@ -52,10 +52,32 @@ var HierarchyView = AmpersandView.extend({
 		});
 		// what we actually want to save is the internal tree object.
 		this.tree = $tree.fancytree('getTree');
+		$tree.on('contextmenu', this.showContextMenu.bind(this));
+	},
+	registerContextOnMenu: function() {
+		var $menu = $('.basicContextContainer');
```

Comment:
can a suitable dom/jQuery element be passed into this function, instead of requiring a selector query every time? e.g. the return value of basicContext.show() below?